### DISCUSSION TITLE
custom (non branded) HTML

### DISCUSSION THREAD

**Discussion by rebred** [sqlpage/SQLPage#731] [Q&A]
most of the pages generated by SQLpage have the following HTML:

`<link rel="icon" href="[/favicon.cbfd4751eaddf743.svg](http://test:8080/favicon.cb4d6221eaddf743.svg)">`
`<link rel="stylesheet" href="[/sqlpage.735cd4232be41811.css](http://test:8080/sqlpage.735cd422be41311.css)">`
`<meta name="generator" content="SQLPage" />`
`<main class="page-body container-xl px-md-5 px-sm-3  mt-3" id="sqlpage_main_wrapper">`

how do we remove the following line
`<meta name="generator" content="SQLPage" />`

how do we change `"sqlpage_main_wrapper"` to `"main_wrapper"` ?

how do we change the word `favicon` with `fav` ?

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage!

  The html generated by SQLPage is entirely customizable. You can replace [the default shell component](https://github.com/sqlpage/SQLPage/blob/main/sqlpage/templates/shell.handlebars) by simply putting a file in `sqlpage/templates/shell.handlebars`

Have a look at [the custom component documentation page](https://sql-page.com/custom_components.sql),
and [this example of an app with fully custom html](https://github.com/sqlpage/SQLPage/tree/main/examples/roundest_pokemon_rating) that does not reuse any of the built-in components.

You can also create a custom shell component without overwriting the existing one by creating a file in `sqlpage/templates/shell_custom.handlebars` and calling it from your code with `select 'shell_custom' as component`
